roomserver/storage/postgres/deltas/20230516154000_drop_reference_sha.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deltas
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+)
+
+func UpDropEventReferenceSHA(ctx context.Context, tx *sql.Tx) error {
+	var count int
+	err := tx.QueryRowContext(ctx, `SELECT count(*) FROM roomserver_events GROUP BY event_id HAVING count(event_id) > 1`).
+		Scan(&count)
+	if err != nil && err != sql.ErrNoRows {
+		return fmt.Errorf("failed to query duplicate event ids")
+	}
+	if count > 0 {
+		return fmt.Errorf("unable to drop column, as there are duplicate event ids")
```

Comment:
This is unfortunate as it would prevent people from upgrading Dendrite. Can we instead mix together the event nids for the affected rows? E.g given:
```
event_id | sha256 | nids
$foo | abc123 | [2,4]
$foo | def456 | [2,3]
```
replace it with:
```
$foo | [2,3,4]
```
?